packages/frontend/src/components/bill/NewBillCard.tsx

```diff
@@ -0,0 +1,207 @@
+import { Button, Textarea } from '@nextui-org/react';
+import DatePicker from 'react-datepicker';
+import 'react-datepicker/dist/react-datepicker.css';
+import React, { useEffect, useState } from 'react';
+import { useApiMutation } from '../../hooks/useApi';
+import { useHouse } from '../../hooks/useHouse';
+
+interface NewBillCardProps {}
+
+interface IHash {
+  [name: string]: string;
+}
+
+interface Bill {
+  title: string;
+  detail: string;
+  dueDate: Date;
+  totalCost: string;
+}
+
+const NewBillCard: React.FC<NewBillCardProps> = () => {
+  const [unixTime, setUnixTime] = useState(0);
+  const [flatmateNum, setFlatmateNum] = useState(6);
+  const [splitCost, setSplitCost] = useState('');
+  const [isEvenlySplit, setIsEvenlySplit] = useState(false);
+  const { users } = useHouse();
+  const [billInfo, setBillInfo] = useState<Bill>({
+    title: ' ',
+    detail: '',
+    dueDate: new Date(),
+    totalCost: '0',
+  });
+
+  const createBill = useApiMutation('/api/v1/house/bills', {
+    method: 'post',
+  });
+
+  let idHash: IHash = {};
+  let costHash: IHash = {};
+  users?.map((user) => {
+    idHash[user.name] = user.firebaseId;
+    costHash[user.name] = '0';
+    return 0;
+  });
+
+  useEffect(() => {
+    setUnixTime(billInfo.dueDate.getTime());
+    setFlatmateNum(users?.length ? users.length : 1);
+    setSplitCost(Number(billInfo.totalCost) / flatmateNum + '');
+  }, [billInfo, flatmateNum, users?.length]);
+
+  const handleEvenlyButton = (e: React.MouseEvent<HTMLButtonElement>) => {
+    setIsEvenlySplit(!isEvenlySplit);
+  };
+
+  const handlePersonalCostChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    costHash[e.target.name] = e.target.value;
+  };
+
+  const handleDoneButton = (e: React.MouseEvent<HTMLButtonElement>) => {
+    let bill = {
+      name: billInfo.title,
+      description: billInfo.detail,
+      due: unixTime,
+      users: [
+        {
+          id: users ? users[0].firebaseId : '0',
+          amount: users
+            ? Number(isEvenlySplit ? splitCost : costHash[users[0].name])
+            : 0,
+          paid: false,
+        },
+      ],
+    };
+
+    users?.map((user, index) => {
+      let newPayment = {
+        id: user.firebaseId,
+        amount: Number(isEvenlySplit ? splitCost : costHash[user.name]),
+        paid: false,
+      };
+      if (index !== 0) {
+        bill['users'].push(newPayment);
+      }
+      return 0;
```

Comment:
just curious what does this do?

Reply:
It was a quick fix, map method needs a return statement, but our process doesn't need it to return anything

Reply:
maybe replace with forEach then? map doesn't make sense if you aren't transforming the data

Reply:
true, I agree, this occur in multiple places as well

Reply:
done